### PR TITLE
[image_picker] use nnbd version of deps to resolve ci failure

### PR DESCRIPTION
**Pull request by cyanglaz**
Landing this to fix CI.
This change will get published with https://github.com/flutter/plugins/pull/3579

### CODE REVIEW

File: packages/image_picker/image_picker/pubspec.yaml [hunk 1]
```diff
@@ -26,8 +26,8 @@ dev_dependencies:
   integration_test:
     path: ../../integration_test
   mockito: ^5.0.0-nullsafety.7
-  pedantic: ^1.8.0
-  plugin_platform_interface: ^1.0.3
+  pedantic: ^1.10.0
+  plugin_platform_interface: ^2.0.0
```
Comment (stuartmorgan) [inline packages/image_picker/image_picker/pubspec.yaml:30]: This needs to be a range.

Comment (cyanglaz) [reply packages/image_picker/image_picker/pubspec.yaml]: oh, forget about it. Done.

**Review by stuartmorgan — APPROVED** [review]
LGTM

**Comment by jmagman** [comment]
This also closed the framework tree.

**Comment by ditman** [comment]
If we're fairly confident this will work, we probably need to merge before pre-submit CI finishes.

At this time of the day, we're heavily throttled both on Cirrus and Firebase some times.

**Comment by ditman** [comment]
Merging in red, to unblock the framework tree.

**Pull request merged**